xirr.gemspec

```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>=2.2.2'
 
-  spec.add_dependency 'activesupport', '~> 6.1'
+  spec.add_dependency 'activesupport', '>= 6.1'
```

Comment:
Maybe we should get rid of this dependency? For such a small library, it is a shame that it depends on so bloated library.

We use activesupport in three places:
1. https://github.com/smartpension/xirr/blob/81e47e7301cb71c0b069e1d601c5ec95df96f44e/lib/xirr/base.rb#L5
To me it looks like it is useless here.
2. https://github.com/smartpension/xirr/blob/81e47e7301cb71c0b069e1d601c5ec95df96f44e/lib/xirr/config.rb#L2
We can replace it with something simple like this:
```ruby
  require 'ostruct'
  def self.config
    @@config ||= OpenStruct.new
  end
```
We don't use `configure` method (provided by active support) at all.
3. We use `to_date` method in tests, eg. https://github.com/smartpension/xirr/blob/81e47e7301cb71c0b069e1d601c5ec95df96f44e/test/test_cashflow.rb#L7
We can use ruby built-in `Date.new(y,m,d)` instead.

@inicklas @MmKolodziej WDYT?

Reply:
i'd love to do that, but unfortunately that's me sidetracking from my main objective getting us to rails 7.

A change like that will require a bit of testing whereas my change is just setting a required minimum of an existing tool.
TLDR;
I'm happy to do that but let's do that as a separate ticket so we can test that code change properly.